src/message.rs

```diff
@@ -207,25 +180,48 @@ impl Transaction {
         attachment_timestamp < current_timestamp
             && current_timestamp - attachment_timestamp < 11 * 60 * 1000
     }
+
+    /// The message's address.
+    pub fn address(&self) -> &Address {
+        unimplemented!()
+    }
+
+    /// The message's hash.
+    pub fn hash(&self) -> &Hash {
+        unimplemented!()
+    }
+
+    /// Gets the absolute value of the transaction.
+    pub fn value(&self) -> Value {
+        let amount = match &self.payload {
+            Payload::SignedTransaction(tx) => tx
+                .unsigned_transaction
+                .outputs
+                .iter()
+                .fold(0, |acc, output| acc + output.amount()),
+            _ => 0,
+        };
+        Value::new(amount.try_into().unwrap(), ValueUnit::I)
+    }
 }
 
-impl PartialEq for Transaction {
+impl PartialEq for Message {
```

Comment:
`bee_transaction::Message` has this trait, so it should be fine to derive it. (and maybe derive other traits too)